vyper/semantics/validation/local.py

```diff
@@ -202,11 +202,21 @@ def __init__(
             ]
             node_list.extend(standalone_self)  # type: ignore
 
+            # Add references to builtin functions reading from the chain's state
+            # TODO Fix circular import
+            from vyper.builtin_functions.functions import BUILTIN_FUNCTIONS
+            builtin_fns = fn_node.get_descendants(vy_ast.Name, {"id": set(BUILTIN_FUNCTIONS)})
+
+            node_list.extend(builtin_fns)  # type: ignore
```

Comment:
why do we need to do this?

Reply:
To add builtin function calls to the list of nodes to check. Or should the check for builtin functions be performed elsewhere?

Reply:
oh for some reason i thought the mutability check was performed under visit_Call. i guess this solution is ok for now; i want to get in #2974 first though